tests/unittests/tests-tweetnacl/tests-tweetnacl.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     unittests
+ * @{
+ *
+ * @file
+ * @brief       tweetnacl NaCl crypto library tests
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <tweetnacl.h>
+#include "embUnit.h"
+#include "tests-tweetnacl.h"
+
+static const char message[] = "0123456789abcdef";
+
+#define MLEN (sizeof(message) + crypto_box_ZEROBYTES)
+
+static unsigned char alice_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char alice_sk[crypto_box_SECRETKEYBYTES];
+static unsigned char bob_pk[crypto_box_PUBLICKEYBYTES];
+static unsigned char bob_sk[crypto_box_SECRETKEYBYTES];
+static unsigned char m[MLEN];
+static unsigned char c[MLEN];
+static const unsigned char n[crypto_box_NONCEBYTES];
+static unsigned char result[MLEN];
+
+static void setUp(void)
+{
+    /* Initialize */
+    random_init(0);
+}
+
+static void tearDown(void)
+{
+    /* Finalize */
+}
+
+static void test_tweetnacl_01(void)
+{
+    int res;
+
+    /* Creating keypair ALICE... */
+    printf("Creating keypair ALICE...\n");
+    crypto_box_keypair(alice_pk, alice_sk);
+    printf("Done.\n");
+
+    /* Creating keypair BOB... */
+    printf("Creating keypair BOB...\n");
+    crypto_box_keypair(bob_pk, bob_sk);
+    printf("Done.\n");
+
+    memset(m, 0, crypto_box_ZEROBYTES);
+    memcpy(m + crypto_box_ZEROBYTES, message, MLEN - crypto_box_ZEROBYTES);
+
+    /* Encrypting using pk_bob... */
+    printf("Encrypting using pk_bob...\n");
+    crypto_box(c, m, MLEN, n, bob_pk, alice_sk);
+
+    memset(result, '\0', sizeof(result));
+
+    /* Decrypting... */
+    printf("Decrypting...\n");
+    res = crypto_box_open(result, c, MLEN, n, alice_pk, bob_sk);
+
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    if (res) {
```

Comment:
Could you replace the printed output with a comparison between message and what was decrypted? You'll need a second buffer...